### PR TITLE
feat: update portal for handling JWT Token API changes

### PR DESCRIPTION
**Pull request by kshitij-k-osmosys**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](https://github.com/OsmosysSoftware/osmo-x/blob/main/CONTRIBUTING.md#-submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [x] I have performed unit testing for the new feature added or updated to ensure the new features added are working as expected.
- [x] I have performed preliminary testing to ensure that any existing features are not impacted and any new features are working as expected as a whole.

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login page`)
- [x] Description has been added
- [x] Related changes have been added (optional)
- [x] Screenshots have been added (optional, may include unit testing screenshots as well)
- [ ] Pending actions have been added (optional)
- [x] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

_Note: Reviewer should ensure that the checklist and description have been populated and followed correctly, and the PR should be merged only after resolving all conversations and verifying that CI checks pass._

---

**Description:**

Update portal for handling JWT Token API changes

**Related changes:**

- Update existing login graphql query to only take token
- Create application model
- Create graphql query and service function for fetching applications
- Update code to get list of applications via api. Create a key value pair between id and name to feed to the dropdown
- Update notification component to add a filter and fetch data for matching applicationId
- Add icons for channel type AWS_SES, SMS_SNS
- Add decodeToken service function for JWT token for future use

**Screenshots:**

image
On initial load when user logins

**Pending actions:**
NA

**Additional notes:**

Commit cef5ee2e6bd79119595c0c84af3d44defecda728 for refactoring code

## Release Notes

- **New Features**
	- Introduced a new GraphQL query `GetApplications` for retrieving application data with pagination and filtering options.
	- Added a new service `ApplicationsService` to manage application data retrieval.
	- Expanded notification channel options with new types: `AWS_SES` for email notifications and `SMS_SNS` for SMS notifications.
	- Enhanced user guidance in the notifications component with prompts for application selection.

- **Bug Fixes**
	- Improved error handling in the notifications component's application fetching logic.

- **Documentation**
	- Updated data models for applications to ensure type safety and consistency.

### CODE REVIEW

File: apps/portal/src/app/views/notifications/notifications.component.ts [hunk 1]
```diff
@@ -85,42 +83,68 @@ export class NotificationsComponent implements OnInit {
...
+  setupConfig(): void {
+    // Simulate async config setup (e.g., API call or local settings)
+    setTimeout(() => {
+      this.getApplications();
+      // Configurations complete
+      this.configSetupComplete = true;
+      // Trigger lazy load manually
+      this.loadNotificationsLazy({ first: 0, rows: this.pageSize });
+    }, 1000); // Adjust timeout or replace with actual async logic
+  }
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply apps/portal/src/app/views/notifications/notifications.component.ts:106]: Tried using await, it still triggers after loadnotification

[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply apps/portal/src/app/views/notifications/notifications.component.ts:106]: .toPromise(); is outdated, also we are already doing this with .subscribe, if not use that

File: apps/portal/src/app/views/notifications/notifications.component.ts [hunk 2]
```diff
@@ -143,16 +153,27 @@ export class NotificationsComponent implements OnInit {
...
+      const userToken = JSON.parse(localStorage.getItem('osmoXUserData')).token;
+      return userToken;
+    } catch (error) {
+      this.messageService.add({
+        key: 'tst',
+        severity: 'error',
+        summary: 'Error',
+        detail: `There was an error fetching the login token: ${error.message}`,
+      });
+      return null;
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply apps/portal/src/app/views/notifications/notifications.component.ts:167]: how to handle potential errors

File: apps/portal/src/common/constants/notification.ts [hunk 3]
```diff
@@ -19,6 +19,8 @@ export const ChannelType = {
   SMS_KAPSYSTEM: 8,
   PUSH_SNS: 9,
   VC_TWILIO: 10,
+  AWS_SES: 11,
```
Comment (xixas) [inline apps/portal/src/common/constants/notification.ts:22]: Are these values new?
Are these something we merged in backend too?

**Review by xixas — CHANGES_REQUESTED** [review]

File: apps/portal/src/common/constants/notification.ts (hunk 3 above)
Comment (kshitij-k-osmosys) [reply apps/portal/src/common/constants/notification.ts:22]: SNS_SMS changes are merged
AWS_SES is yet to be merged

File: apps/portal/src/app/views/notifications/notifications.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply apps/portal/src/app/views/notifications/notifications.component.ts]: Renamed function. We have error handling in the function now

File: apps/portal/src/app/views/notifications/notifications.component.ts [hunk 4]
```diff
@@ -143,16 +164,38 @@ export class NotificationsComponent implements OnInit {
...
+      throw new Error('User data not found in localStorage');
+    } catch (error) {
+      this.messageService.add({
+        key: 'tst',
+        severity: 'error',
+        summary: 'Error',
+        detail: `There was an error fetching the login token: ${error.message}`,
+      });
+      return null;
+    }
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply apps/portal/src/app/views/notifications/notifications.component.ts]: Renamed function. We have error handling in the function now

**Review by xixas — APPROVED** [review]

**Pull request merged**